docs/source/references/config.md

```diff
@@ -120,7 +107,7 @@ The following is the default configuration used by Iroha.
 
 `BlockSynchronizer` configuration
 
-Has type `block_sync::Configuration`. Can be configured via environment variable `IROHA_BLOCK_SYNC`
+Has type `Option<block_sync::ConfigurationProxy>`. Can be configured via environment variable `IROHA_BLOCK_SYNC`
```

Comment:
We should add an explanatory note to the header, saying that the `Option<bool>` config variables will fall back to a default value if not provided, while the `bool` value must be specified explicitly. Ideally all `Option<…>` segments should point to that (foot)note.

Reply:
Done a very extensive one.